db/queries.py

```diff
@@ -1,46 +1,89 @@
+from __future__ import annotations
+
 from db.models import Message, User, Chat
 from django.db.models import Q, Count, F
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(
+        Message.objects.select_related("user", "chat")
+        .filter(text__icontains=word)
+    )
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return list(
+        Chat.objects.filter(
+            title__startswith="Untitled"
+        )
+    )
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(
+        User.objects.filter(message__sent__year="2015").values_list(
+            "first_name", "last_name"
+        )
+    )
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(
+        Chat.objects.filter(message__sent__year__gt="2020")
+    )
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(
+        Message.objects.select_related("user", "chat")
+        .filter(text__contains=F("user__first_name"))
```

Comment:
You don’t need `select_related `here as well

Reply:
The same